### PR TITLE
Changes to support new opx testbed

### PR DESCRIPTION
**Pull request by Tejaswi-Goel**
Signed-off-by: Tejaswi Goel <tejaswi_goel@dell.com>

### CODE REVIEW

File: playbook/Forwarding/roles/span_mirroring/tasks/setup.yml [hunk 1]
```diff
@@ -1,5 +1,28 @@
...
+        - { pg: "{{port_group1}}", mode: "{{splitport}}" }
+        - { pg: "{{port_group2}}", mode: "{{mode}}" }
+    register: result
+    when: (inventory_hostname == "TR")
+  - name: Check the fanout port exists on DUT and TR
+    command: bash -c "ip link show {{server_int}}"
+    register: result 
+    when: (inventory_hostname == "DUT") or (inventory_hostname == "TR")
+  - debug:
+      var: result
```
Comment (rakeshdatta) [inline playbook/Forwarding/roles/span_mirroring/tasks/setup.yml]: make this: 
- name: STDOUT the ip link show output
  debug:
       msg: "{{result.stdout}}"
  ignore_errors: yes


Also, dont we need a when clause here?

File: playbook/L2/roles/l2_stp/tasks/setup.yml [hunk 2]
```diff
@@ -1,5 +1,28 @@
...
+    with_items:
+        - { pg: "{{port_group1}}", mode: "{{splitport}}" }
+        - { pg: "{{port_group2}}", mode: "{{mode}}" }
+    register: result
+    when: (inventory_hostname == "TR")
+  - name: Check the fanout port exists on DUT and TR
+    command: bash -c "ip link show {{server_int}}"
+    register: result 
+    when: (inventory_hostname == "DUT") or (inventory_hostname == "TR")
+  - debug:
```
Comment (rakeshdatta) [inline playbook/L2/roles/l2_stp/tasks/setup.yml]: same comment as for other playbooks

File: playbook/L2/roles/l2mac/tasks/setup.yml [hunk 3]
```diff
@@ -1,16 +1,32 @@
...
+    with_items:
+        - { pg: "{{port_group1}}", mode: "{{splitport}}" }
+        - { pg: "{{port_group2}}", mode: "{{mode}}" }
+    register: result
+    when: (inventory_hostname == "TR")
+  - name: Check the fanout port exists on DUT and TR
+    command: bash -c "ip link show {{server_int}}"
+    register: result 
+    when: (inventory_hostname == "DUT") or (inventory_hostname == "TR")
+  - debug:
```
Comment (rakeshdatta) [inline playbook/L2/roles/l2mac/tasks/setup.yml]: same comment as for other playbooks

File: playbook/L2/roles/l3_intra_inter/tasks/setup.yml [hunk 4]
```diff
@@ -1,16 +1,32 @@
...
+    with_items:
+        - { pg: "{{port_group1}}", mode: "{{splitport}}" }
+        - { pg: "{{port_group2}}", mode: "{{mode}}" }
+    register: result
+    when: (inventory_hostname == "TR")
+  - name: Check the fanout port exists on DUT and TR
+    command: bash -c "ip link show {{server_int}}"
+    register: result 
+    when: (inventory_hostname == "DUT") or (inventory_hostname == "TR")
+  - debug:
```
Comment (rakeshdatta) [inline playbook/L2/roles/l3_intra_inter/tasks/setup.yml]: same comment as for other playbooks

File: playbook/L3/roles/l3-basic-functionality/tasks/setup.yml [hunk 5]
```diff
@@ -1,5 +1,28 @@
...
+    with_items:
+        - { pg: "{{port_group1}}", mode: "{{splitport}}" }
+        - { pg: "{{port_group2}}", mode: "{{mode}}" }
+    register: result
+    when: (inventory_hostname == "TR")
+  - name: Check the fanout port exists on DUT and TR
+    command: bash -c "ip link show {{server_int}}"
+    register: result 
+    when: (inventory_hostname == "DUT") or (inventory_hostname == "TR")
+  - debug:
```
Comment (rakeshdatta) [inline playbook/L3/roles/l3-basic-functionality/tasks/setup.yml]: same comment as for other playbooks

File: playbook/L3/roles/l3-basic-functionality/tasks/l3-basic-functionality.yml [hunk 6]
```diff
@@ -235,7 +231,7 @@
   - debug:
       var: result
   - name: TC5 - Verify bridge does not exist
-    command: bash -c "opx-show-vlan --id {{vlanid}}"
```
Comment (rakeshdatta) [inline playbook/L3/roles/l3-basic-functionality/tasks/l3-basic-functionality.yml]: why the vlanid is removed?

File: playbook/L3/roles/Routing_IPv6/tasks/Routing_IPv6_base_functionality.yml [hunk 7]
```diff
@@ -90,10 +90,6 @@
     command: "ip link set dev {{server1_port}} down"
     register: up_interface_state_server1
     when: inventory_hostname == "Server1"
-  - name: Revert fanned out port connected to server
```
Comment (rakeshdatta) [inline playbook/L3/roles/Routing_IPv6/tasks/Routing_IPv6_base_functionality.yml]: Dont remove this

File: playbook/L3/roles/Routing_IPv6/tasks/Routing_IPv6_base_functionality.yml [hunk 8]
```diff
@@ -309,15 +305,12 @@
         - "{{test_int1}}"
         - "{{server_int}}"
     register: set_interface_state_dut
+    ignore_errors: yes
     when: inventory_hostname == "DUT"
   - name: TC2 - set interface state Down Server1
     command: "ip link set dev {{server1_port}} down"
     register: down_interface_state_server1
     when: inventory_hostname == "Server1"
-  - name: Revert fanned out port connected to server
```
Comment (rakeshdatta) [inline playbook/L3/roles/Routing_IPv6/tasks/Routing_IPv6_base_functionality.yml]: Dont remove this

File: playbook/L3/roles/Routing_IPv6/tasks/Routing_IPv6_base_functionality.yml [hunk 9]
```diff
@@ -543,7 +536,3 @@
     command: "ip link set dev {{server_link}} down"
     register: up_interface_state_server1
     when: inventory_hostname == "Server1"
-  - name: TC3-Revert fanned out port connected to server
```
Comment (rakeshdatta) [inline playbook/L3/roles/Routing_IPv6/tasks/Routing_IPv6_base_functionality.yml]: Dont remove this

File: playbook/L3/roles/l3-basic-functionality/tasks/l3-basic-functionality.yml [hunk 10]
```diff
@@ -93,10 +93,6 @@
     command: bash -c "echo 'Start Cleanup for TC2'"
   - name: TC2 - Bring down the fanout interface on DUT
     command: bash -c "ip link set {{server_int}} down; ip addr flush dev {{server_int}}"
-  - name: TC2 - Disable the port fanout configuraion (fanin the port)
```
Comment (rakeshdatta) [inline playbook/L3/roles/l3-basic-functionality/tasks/l3-basic-functionality.yml:96]: Dont remove this

File: playbook/L3/roles/l3-basic-functionality/tasks/l3-basic-functionality.yml [hunk 11]
```diff
@@ -500,7 +500,7 @@
       var: result
     when: (inventory_hostname == "DUT") 
   - name: TC9 - Configure IPv4 address to bridge {{bridge}} on TR
-    command: bash -c "ip addr flush dev {{bridge}}; ip addr add {{IPAddr2}}/{{prefixlen}} dev {{bridge}}; ifconfig {{bridge}}"
+    command: bash -c "ip addr flush dev {{bridge}} && ip addr add {{IPAddr2}}/{{prefixlen}} dev {{bridge}} && ifconfig {{bridge}} && ping -c 3 {{IPAddr1}}"
```
Comment (rakeshdatta) [inline playbook/L3/roles/l3-basic-functionality/tasks/l3-basic-functionality.yml:506]: Make sure that there are proper handling when ping fails

**Review by rakeshdatta — CHANGES_REQUESTED** [review]
- Plz address the comments
- Plz attach the logs

File: playbook/L3/roles/l3-basic-functionality/tasks/l3-basic-functionality.yml (hunk 6 above)
Comment (Tejaswi-Goel) [reply playbook/L3/roles/l3-basic-functionality/tasks/l3-basic-functionality.yml]: root@OPX:~# opx-show-vlan --id 100   <-- error seen in output (in current release) 
Traceback (most recent call last):
  File "/usr/bin/opx-show-vlan", line 199, in <module>
    obj.show_vlans(name)
  File "/usr/bin/opx-show-vlan", line 128, in show_vlans
    self.print_vlan(obj[0])
IndexError: list index out of range

root@OPX:~# opx-show-vlan
opx-show-vlan: No VLAN information found

File: playbook/L3/roles/l3-basic-functionality/tasks/l3-basic-functionality.yml [hunk 12]
```diff
@@ -216,7 +215,7 @@
     command: bash -c "opx-show-route | grep {{route0}}"
     register: result
     failed_when: result.rc == 0   
-  tags: ['Verify user can add modify and delete ipv4 address to a vlan_interface','sanity']
```
Comment (Tejaswi-Goel) [inline playbook/L3/roles/l3-basic-functionality/tasks/l3-basic-functionality.yml:219]: Removing sanity from this test case due to the following error:
oot@OPX:~# opx-show-vlan --id 100 <-- error seen in output (in current release)
Traceback (most recent call last):
File "/usr/bin/opx-show-vlan", line 199, in 
obj.show_vlans(name)
File "/usr/bin/opx-show-vlan", line 128, in show_vlans
self.print_vlan(obj[0])
IndexError: list index out of range

Expected output: _opx-show-vlan: No VLAN information found_

File: playbook/L3/roles/l3-basic-functionality/tasks/l3-basic-functionality.yml (hunk 11 above)
Comment (Tejaswi-Goel) [reply playbook/L3/roles/l3-basic-functionality/tasks/l3-basic-functionality.yml:506]: Once the ping fails (rc =1) test case should fail , rescue section and always section will be executed,  no need of extra code for ping failing handling.

**Comment by Tejaswi-Goel** [comment]
Test logs

> * Plz address the comments
> * Plz attach the logs

**Comment by Tejaswi-Goel** [comment]
> * Plz address the comments
> * Plz attach the logs

Logs: http://10.11.177.93:8080/job/sanity/9/consoleFull

**Review by rakeshdatta — APPROVED** [review]

**Pull request merged**